### PR TITLE
Trips connect backend

### PR DESCRIPTION
**Pull request by Takahiro9-Murakami9**

### CODE REVIEW

File: context/AuthContext.js [hunk 1]
```diff
@@ -2,11 +2,12 @@ import AsyncStorage from "@react-native-async-storage/async-storage";
 import React, { createContext, useEffect, useState } from "react";
 import API_URL from "../config";
 
+
 export const AuthContext = createContext();
 
 export function AuthProvider({children}) {
   const [isLoading, setIsLoading] = useState(false);
-  const [userToken, setUserToken] = useState(null);
+  const [userToken, setUserToken] = useState();
```
Comment (YJK-7) [inline context/AuthContext.js]: useState(null)

File: context/AuthContext.js [hunk 2]
```diff
@@ -72,9 +77,12 @@ export function AuthProvider({children}) {
 
   const isLoggedIn = async () => {
     try{
+      // if (!userToken) {//new line
+      //   setIsLoading(false)// new line
+      // }else // new line
       setIsLoading(true)
       let userTokenStored = await AsyncStorage.getItem('userToken');
```
Comment (YJK-7) [inline context/AuthContext.js:81]:
```
if(userTokenStored === null){
        setIsLoading(false);
        return
} else { the rest of the code in isLoggedIn}
```

File: context/AuthContext.js [hunk 3]
```diff
@@ -86,17 +94,20 @@ export function AuthProvider({children}) {
...
+      console.log(userData);
       setIsLoading(false);
     } catch (e) {
       console.log(`Login Error: ${e}`)
     }
   }
 
   useEffect(() => {
-    isLoggedIn()
+    if (userToken) {
```
Comment (YJK-7) [inline context/AuthContext.js]: don't need if here

File: context/InfoContext.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
 import API_URL from "../config";
+import moment from 'moment';
 
 export const InfoContext = createContext();
 
 export function InfoProvider({children}) {
   const [trips, setTrips] = useState(null);//all trips for user
   const [tripEvents, setTripEvents] = useState(null);//trip events for one trip
   const [tripid, setTripid] = useState(null);//trip id
+  const [userid, setUserid] = useState(null);//user id
```
Comment (YJK-7) [inline context/InfoContext.js]: don't need?

File: package.json [hunk 5]
```diff
@@ -14,15 +14,16 @@
     "@react-navigation/bottom-tabs": "^6.5.2",
     "@react-navigation/native": "^6.1.1",
     "@react-navigation/native-stack": "^6.9.7",
-    "expo": "~47.0.8",
+    "expo": "^47.0.0",
     "expo-constants": "~14.0.2",
     "expo-status-bar": "~1.4.2",
     "formik": "^2.2.9",
+    "jwt-decode": "^3.1.2",
```
Comment (YJK-7) [inline package.json:21]: don't need

File: screens/AddTrip.js [hunk 6]
```diff
@@ -1,8 +1,12 @@
 import { Formik } from 'formik';
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
+import { AuthContext } from "../context/AuthContext";
+import { InfoContext } from '../context/InfoContext';
 import { StyleSheet, Button, TextInput, View, TouchableOpacity } from 'react-native';
 import { Octicons, Ionicons, Entypo } from '@expo/vector-icons';
 import DateTimePicker from '@react-native-community/datetimepicker';
+import moment from 'moment';
+import jwt_decode from "jwt-decode";
```
Comment (YJK-7) [inline screens/AddTrip.js]: don't need

File: screens/AddTrip.js [hunk 7]
```diff
@@ -11,20 +15,37 @@ import {
     Colors, 
 } from '../styles/styles';
 
+
 const {brand, darkLight} = Colors;
 
-export default function AddTrip(params) {
+export default function AddTrip({setModalOpen}) {
   const [show, setShow] = useState(false);
   const [date, setDate] = useState(new Date(2023, 0, 1));
```
Comment (YJK-7) [inline screens/AddTrip.js]: new Date()
this will set the default date to today's date

File: screens/AddTrip.js [hunk 8]
```diff
@@ -11,20 +15,37 @@ import {
...
   const [startDate, setStartDate] = useState();
   const [endDate, setEndDate] = useState();
 
+  const { userToken } = useContext(AuthContext);
```
Comment (YJK-7) [inline screens/AddTrip.js]: don't need

File: screens/AddTrip.js [hunk 9]
```diff
@@ -11,20 +15,37 @@ import {
...
+  // console.log("🍶",userToken);
+  const getId = jwt_decode(userToken).userid;
```
Comment (YJK-7) [inline screens/AddTrip.js]: don't need

File: screens/AddTrip.js [hunk 10]
```diff
@@ -47,8 +68,11 @@ export default function AddTrip(params) {
       <Formik
         initialValues={{ startDate: '', endDate: '', userid: '', name: '' }}
         onSubmit={(values) =>{
-          values = {...values, startDate: startDate, endDate: endDate}
-          console.log(values);
+          values = {...values, userid: getId, startDate: moment(startDate).format("YYYY-MM-DD"), endDate: moment(endDate).format("YYYY-MM-DD")}
```
Comment (YJK-7) [inline screens/AddTrip.js]: don't need to send in userid

File: context/AuthContext.js [hunk 11]
```diff
@@ -72,9 +77,12 @@ export function AuthProvider({children}) {
 
   const isLoggedIn = async () => {
```
Comment (YJK-7) [inline context/AuthContext.js:74]: could you also add `// only when user logged before and quite app` above `isLoggedIn` function so that we (aka I ) don't get confused about its function?

File: context/InfoContext.js [hunk 12]
```diff
@@ -9,41 +10,8 @@ export function InfoProvider({children}) {
...
-
-  // Trips
-  const tripData = { //using userData.id
-    2:[
-      {tripName: 'Tokyo', id:1, startDate: '2022-12-20', endDate: '2022-12-22', userid: 2},
-      {tripName: 'Osaka', id:2, startDate: '2022-12-04', endDate: '2022-12-06', userid: 2},
-      {tripName: 'Boston', id:3, startDate: '2023-01-04', endDate: '2023-01-06', userid: 2},
-    ]
-  }
-
```
Comment (calss0t) [inline context/InfoContext.js:25]: Why did you delete tripData?

**Review by calss0t — APPROVED** [review]
Looks good

**Review by YJK-7 — APPROVED** [review]
👍
